### PR TITLE
person dispatcher and gym exercise added

### PR DESCRIPTION
**Pull request by cadin7**

### CODE REVIEW

File: src/main/java/ro/fasttrackit/hw3/gym/GymMain.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public static void main(String[] args) throws IOException {
+        new ReportGenerator(new InMemorySubscriptionReader(), new ReportWriter()).generateReport();
+
+        printOperations();
+    }
+
+    private static void printOperations() {
+        final var arnySchwarzy = new Member(1L, "Arny Schwarzy", LocalDate.of(1947, 7, 30));
+        final var stony = new Member(2L, "A Stone", LocalDate.of(1972, 5, 2));
+        final var beluci = new Member(3L, "Kati Bela", LocalDate.of(1979, 5, 4));
```
Comment (sinkobela) [inline src/main/java/ro/fasttrackit/hw3/gym/GymMain.java:25]: Actually... nice research =))

File: src/main/java/ro/fasttrackit/hw3/gym/GymMain.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        final var stonySub = new Subscription(stony, 20);
+        final var belaSub = new Subscription(beluci, 5);
+
+        final var gym = new Gym(Set.of(arnySub, stonySub, belaSub));
+
+        getMembersInfo(gym);
+
+        System.out.println("The MIN age is: " + gym.getMinimumAge());
+        System.out.println("The AVG age is: " + gym.getAverageAge());
+        System.out.println("The MAX age is: " + gym.getMaximumAge());
```
Comment (sinkobela) [inline src/main/java/ro/fasttrackit/hw3/gym/GymMain.java:37]: Those printings should be extracted to a method.

File: src/main/java/ro/fasttrackit/hw3/gym/model/Gym.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public int getMaximumAge() {
+        return getAgeStatistics().getMax();
+    }
+
+    public String getMemberInfo(String name) {
+        return subscriptions.stream()
+                .filter(subscription -> subscription.getMember().name().contains(name))
+                .map(Subscription::toString)
+                .findFirst()
```
Comment (sinkobela) [inline src/main/java/ro/fasttrackit/hw3/gym/model/Gym.java:56]: The findFirst shouldn't be executed before the mapping ?

**Review by sinkobela — APPROVED** [review]

File: src/main/java/ro/fasttrackit/hw3/gym/model/Gym.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                .filter(subscription -> subscription.getMember().id() == member.id())
+                .findFirst();
+    }
+
+    private IntSummaryStatistics getAgeStatistics() {
+        return subscriptions.stream()
+                .map(Subscription::getMember)
+                .map(Member::birthdate)
+                .mapToInt(this::getYearsFromNow)
+                .summaryStatistics();
```
Comment (markvaradi94) [inline src/main/java/ro/fasttrackit/hw3/gym/model/Gym.java:71]: Cool stuff, didn't know it existed :)

**Review by markvaradi94 — APPROVED** [review]
GG bro, looks really good <3

**Pull request merged**